submit_cgap/submission.py

```diff
@@ -641,9 +642,101 @@ def do_uploads(upload_spec_list, auth, folder=None, no_query=False, subfolders=F
+
+ALL_S3_STORAGE_CLASSES = [
+    'STANDARD', 'REDUCED_REDUNDANCY', 'STANDARD_IA', 'ONEZONE_IA', 'INTELLIGENT_TIERING',
+    'GLACIER', 'DEEP_ARCHIVE', 'OUTPOSTS', 'GLACIER_IR',
+]
+
+AVAILABLE_S3_STORAGE_CLASSES = [
+    'STANDARD', 'STANDARD_IA', 'INTELLIGENT_TIERING'
+]
+
+CGAP_S3FS_MAPPING_PATTERN = re.compile(r"^([^:]+):(.+)$")
+
+
+def check_s3fs_mapped_filename(filename, *, s3):
+    """
+    Determines whether a given filename is believed to be mapped by s3fs.
+
+    :param filename: the filename to check
+    :param s3: an s3 client.
+    :return:
+    """
+    s3 = s3 or boto3.client('s3')
+    upload_buckets = os.environ.get("CGAP_S3FS_UPLOAD_BUCKETS")
+    upload_dir = os.environ.get("CGAP_S3FS_UPLOAD_DIR")
+    if not upload_buckets or not upload_dir:
+        # We're not using S3FS mapping, so we have no warnings to show.
+        return None
+    mapped_dir = upload_dir.rstrip('/')
+    pattern = f"^(?:{re.escape(mapped_dir)}|{re.escape(os.path.expanduser(mapped_dir))})/(.*)$"
+    m = re.match(pattern, filename)
+    if not m:
+        # Some files might not match our mapping.
+        return None
+    mapped_key = m.group(1)
+    candidates = bash_enumeration(upload_buckets)
+    for mapped_bucket in candidates:
+        try:
+            s3.head_object(Bucket=mapped_bucket, Key=mapped_key)  # an error means we're failing
+            return mapped_bucket, mapped_key
+        except Exception:
+            pass
+    else:
+        # No suitable match found
+        return None
+
+
+STRING_LIST_SEPARATORS = str.maketrans("\n\t,", "   ")
+
+
+def bash_enumeration(string_list):
+    """
+    For use to parse a bash variable value that purports to contain a list.
+    The items in the list can be separated by commas, spaces, or newlines
+    (and so the elements may not contain those characters).
+
+    :param string_list: a string representing a list of items separated by commas, spaces, or newlines.
+    :return: the items in the list, as strings
+    """
+    return [x for x in string_list.translate(STRING_LIST_SEPARATORS).split(" ") if x]
+
+
+def maybe_show_s3fs_warnings(filename, *, s3=None):
+    """
+    If the indicated filename is mapped bucket by s3fs to an s3 bucket/key,
+    this checks whether warning about its storage class is warranted and, if it is, presents such a warning.
+    """
+    s3 = s3 or boto3.client('s3')  # noQA
+    mapped_bucket_and_key = check_s3fs_mapped_filename(filename, s3=s3)
+    if not mapped_bucket_and_key:
+        return False
+    # If we get to here, we have a plausibly mapped filename that we might want to warn about.
+    mapped_bucket, mapped_key = mapped_bucket_and_key
+    try:
+        metadata = s3.head_object(Bucket=mapped_bucket, Key=mapped_key)
+        storage_class = metadata['StorageClass']
+        if storage_class not in AVAILABLE_S3_STORAGE_CLASSES:
+            show(f"The file {filename} is mapped via S3FS to {storage_class} storage.")
```

Comment:
You might consider expanding this to include more information. I'm not sure we should assume this statement alone is enough to convey the problem (and the solution). I would just say something like: 

```
show(f"The file {filename} is mapped via S3FS to {storage_class} storage."
      " Use awscli or equivalent to restore this file to a storage class that it can be retrieved from: {AVAILABLE_S3_STORAGE_CLASSES}")
````